src/common/kube-helpers.ts

```diff
@@ -151,27 +151,43 @@ export function getNodeWarningConditions(node: V1Node) {
 }
 
 /**
- * Validates kubeconfig supplied in the add clusters screen. At present this will just validate
- * the User struct, specifically the command passed to the exec substructure. 
- */ 
-export function validateKubeConfig (config: KubeConfig) {
+ * Validates Context, User and Cluster sructs in given kubeconfig. Additionally this will validate
+ * the command passed to the exec substructure.
+ */
+export function validateKubeConfig (config: KubeConfig, contextName: string, validationOpts?: { validateCluster?: boolean, validateUser?: boolean, validateExec?: boolean}) {
```

Comment:
Can you extract the this out to `ValidationOptions`?

Reply:
And provided a default `{}` for `validationOpts`

Reply:
Done